merklize/merklize.go

```diff
@@ -1083,16 +1083,110 @@ func valueToHash(h Hasher, datatype string, value any) (*big.Int, error) {
 }
 
 // only supported xsd types.
-func convertAnyToString(value any) (str string, err error) {
+func convertAnyToString(value any, datatype string) (str string, err error) {
+
+	if datatype == ld.XSDDouble {
+		switch v := value.(type) {
+		case string:
+			f, err := strconv.ParseFloat(v, 64)
+			if err != nil {
+				return "", err
+			}
+			return ld.GetCanonicalDouble(f), nil
+		case int:
+			return intToXSDDoubleStr(v)
+		case int8:
+			return intToXSDDoubleStr(v)
+		case int16:
+			return intToXSDDoubleStr(v)
+		case int32:
+			return intToXSDDoubleStr(v)
+		case int64:
+			return intToXSDDoubleStr(v)
+		case uint:
+			return uintToXSDDoubleStr(v)
+		case uint8:
+			return uintToXSDDoubleStr(v)
+		case uint16:
+			return uintToXSDDoubleStr(v)
+		case uint32:
+			return uintToXSDDoubleStr(v)
+		case uint64:
+			return uintToXSDDoubleStr(v)
+		}
+	}
+
 	switch v := value.(type) {
-	case string, float64, float32, int64, int32, int16, int8, int, bool:
+	case float64:
```

Comment:
Consider the next case.
We have JSON document (vp like example):
```
{
   age: 30,
}
```
We are parsing JSON and get age. Age is float64, since all numbers in JSON is float64. But out schema has "xsd:integer" tipe for `age`. In this case we will get incorrect hash. Since merlize lib will take hahs from `hash(30)` but this code will take `hash(floatConinise(30))`.

Could you add a test for integers and compare two hashes from these code and hash after merklized.